lib/query-emitter.js

```diff
@@ -90,28 +95,35 @@ QueryEmitter.prototype._update = function(op) {
   // that removed the doc from the query to cause the client-side computed
   // list to update.
   if (this.ids.indexOf(id) !== -1) {
-    // Note that this op is not projected or sanitized yet. It's possible for a
-    // query to filter on a field that's not in the projection. skipPoll checks
-    // to see if an op could possibly affect a query, so it should get passed
-    // the full op. The onOp listener function must call backend.sanitizeOp()
-    this.onOp(op);
+    var emitter = this;
+    pollCallback = function(err) {
```

Comment:
Nice. When I was looking at this, I was thinking of making a separate function for the query-poll code below, but this works just as well!